### ISSUE TITLE
Issues with my S2 since the migration to the 0.6.3 patch

### ISSUE THREAD

**Issue by goudii** [blopker/superduper#42]
Hi,

The app stopped to work like it used to do with the previous patch (0.5) since the app updated itselft. Sometimes the bike won’t pair at all, or pair but the throttle will stay disabled. Sometimes it is insanely long (5mins to pair) and i have to restart the app several times. I also felt that the torque was greatly decreased. When i constated these issues i disabled the auto update and downgraded to the previous patch (the 0.5) and my bike started to work again, but yesterday i was forced to go to the new patch with an error telling me that the previous version was not available anymore. Then the bike started to troll me again. Can you please explain how to fix my bike so i can use it again, or give access to the previous patch? thanks🙏🏻

The app : version 0.6.3
the device used : Iphone 12 mini
the bike : SUPER73 S2 EU version

thanks a lot🙏🏻

**Comment by blopker** [comment]
Hey! Sorry to hear the app is giving you issues. There may be a few problems happening at the same time though. 

The first thing to try is to go to your bike settings in SD and make sure it's in the right region, EU. This might be the problem of the bike acting funny when the app finally connects.

The second, make sure GPS is enabled. I've heard some devices need to make Bluetooth connect better. 

Let me know if any of that helps!

**Comment by mbaran5** [comment]
I can report the same behavior with my US S2. I have to either force close the app or turn my bike on/off to get it to pair.

**Comment by goudii** [comment]
After some tests the torque is back, but still really slow to connect. However, i just bought a RX (eu version) and the app is totally ineffective on the new bike. I can’t turn on the lights or change between modes/pedal assist. I am forced to use the official app in order to make it work. I tried to uninstall the app, reinstall it, delete the existing S2 in order to only have the new RX registered, but it doesn’t affect the new bike at all. Do you know how i can fix it? tried to swap the area where the bike is located, US mode, EU mode etc. Nothing seems to work. Do you know how i can make it work? Thank you in advance.

**Comment by blopker** [comment]
This typically happens if the bike is on an old firmware. Does the official app say there's an update?

**Comment by mbaran5** [comment]
I am skeptical to use the official app since the last big update (the double update that removed the permanent off road + added torque boost) it bricked the bike and I got a one time grace replacement (and I sent you my broken one to tinker with). 

I can check, but I do know the new controller had the latest firmware as of ~june 2023.

**Comment by blopker** [comment]
Oh hey! As far as I know, there hasn't been another firmware update since the tourque update, but so far 100% of the people with the problem you've described has been due to an outdated firmware. Other than that, make sure to check the FAQ: https://github.com/blopker/superduper?tab=readme-ov-file#the-app-wont-connect-to-my-bike

**Comment by mbaran5** [comment]
So, it finds it immediately. I just click Connect and it says "Connecting..." and that takes a while before it does anything. Sometimes it doesn't work so I force close the app and power cycle the bike - that usually does it.

Is there a way to enable debug logs in the app? Happy to go try and send em over if it helps. As stated by OP, I didn't haave this problem until the app redesign.

**Comment by blopker** [comment]
The app is a pretty standard Flutter app, which has a relatively (for mobile apps) easy setup. If you'd like to take a look at the logs you'll need to grab the source and make a debug build. But yeah, some people report a better connection and some worse after the update. I'm still looking for ways to improve though.

**Comment by goudii** [comment]
> This typically happens if the bike is on an old firmware. Does the official app say there's an update?

The bike is updated with the official app, the firmwares are up to date. It is still not working 😰

**Comment by goudii** [comment]
update : it works! An update appeared on the official app, i guess it was the torque update from 2023. Everything connect fine now, and it’s really fast to connect to my RX (as blopker said before, it depends of the bike.) I now have another issue : my bike keeps shutting off while accelerating, as if the torque update was too much for the controller. When im starting from a full stop, weither it’s by peddling or with throttle, I cannot fully engage or my bike will stop for let’s say 5 seconds, then will go again. It’s also very hard to hit 33mph now, the bike will barely go over 28mph.

**Comment by Bamuel** [comment]
Thanks for the inside info @goudii,

I'm assuming your bike battery is near 100% charge when its cutting out?

**Comment by goudii** [comment]
> its cutting out?

yes exactly. The amount of power losses dicrease as much as I drain my battery. The morning it shutdown at almost every stops, the afternoon when i come back from work it shutdown every 2. I’ve told my reseller about the issue, he told me that it was due to the big torque update, and that it was a special firmware for EU users to give more torque to the EPAC mode, but it also give way too much torque for the controller. I barely move my feets and it goes like crazy. He told me that nothing could be done and we couldn’t downgrade to the previous firmware. I’m glad I didn’t update my S2. It would be nice to be able to modify the power imput, or to chose what firmware we want to use on the duper app. But i guess my brand new RX is fucked until a new firmware is available, what a pity for a 4k bike…
